gui/src/app/RunPanel/RunPanel.tsx

```diff
@@ -39,62 +45,90 @@ const RunPanel: FunctionComponent<RunPanelProps> = ({
     sampler.cancel();
   }, [sampler]);
 
-  if (!sampler)
-    return <div className="RunPanelPadded">Stan model not compiled</div>;
+  const { compile, compileMessage, compileStatus, validSyntax } =
+    useContext(CompileContext);
+
+  const { data: projectData } = useContext(ProjectContext);
 
   if (!dataIsSaved) {
     return <div className="RunPanelPadded">Data not saved</div>;
   }
   return (
     <div className="RunPanel">
       <div className="RunPanelPadded">
-        <div>
-          <Button
-            variant="contained"
-            color="success"
-            onClick={handleRun}
-            disabled={runStatus === "sampling" || runStatus === "loading"}
-          >
-            run sampling
-          </Button>
-          &nbsp;
-          {runStatus === "sampling" && (
+        {compileStatus === "compiled" ? (
```

Comment:
This now has enough `? :` and `&&`s in it that I find it pretty difficult to follow. Do you think we could split this up either into a few small components or something like a `content` variable with some if statements before the return?

Reply:
Done